### ISSUE TITLE
How to contribute to premake

### ISSUE THREAD

**Issue by redorav** [premake/premake-core#1017]
I'm trying to debug the lua scripts premake comes with, in an attempt to actually be useful and perhaps fix a bug here or there. I've managed to compile and debug the c code, but debugging Lua seems a bit more involved. I've tried both Decoda and Babelua following the guides in the wiki with no success. I point the executable at a debug build of premake, add the --scripts commandline but I get the following error message after it successfully runs: "Warning 1000: Lua functions were not found during debugging session", and it doesn't stop at any breakpoint I set.

How do you debug premake? @samsinsane @starkos @tvandijck If I could get it to work it would be awesome, I think I'll update the guide as it's a bit incomplete at the moment IMO.

**Comment by xgalaxy** [comment]
I havn't tried debugging premake before I believe you have to use a premake executable compiled with some debugging options enabled.

**Comment by redorav** [comment]
@xgalaxy thanks for replying, I read through that same document trying to get it to work, I also looked in the Google developer's forum to the person that originally got this working and put the steps in the wiki. With Babelua, I got to the point where I pointed it to Lua5.1.exe and ran _premake_main.lua instead of pointing it to premake.exe and got it to "work". However premake Lua scripts use functionality that was introduced in Lua 5.2 (like bit shifting) so at one point it just stops running. However, running Babelua with Lua 5.2 crashes and with 5.3 (what premake uses) it says it found no Lua functions, so I suspect at the point where premake upgraded the embedded Lua version is when the process described in the Wiki stopped working.

**Comment by starkos** [comment]
I've never managed to get a debugger running myself, and just use old-school printf() style. It would be great to get one working, and I'd be happy to approve any pull requests needed.

**Comment by tvandijck** [comment]
Yup, same here.... printf and logs to text files.

**Comment by samsinsane** [comment]
Same, I thought I was the only one using printf, haha. If you get a debugger working, that would be pretty cool to see and use.

**Comment by redorav** [comment]
Ok, so quick update. I've managed to get premake running with the ZeroBrane debugger, it seems it's the only debugger that supports Lua 5.3 these days. I've had to build premake with luasocket as a static lib as premake loads the embedded Lua and ZeroBrane expects it to come from a dll. After the integration it's actually pretty straightforward to hook up and start debugging and I can update the guide to show the steps required to get it running.

I can try cleaning up the code over the weekend and creating a PR if you like to review the code?
